syntax = "proto3";
import "google/protobuf/timestamp.proto";

message State {
  string type = 1;
  string name = 2;
}

message Faction {
  int64 id = 1;
  string name = 2;
  string allegiance = 3;
  bool controlling = 4;
  string government = 5;
  float influence = 6;
  string happiness = 7;
  bool player_faction = 8;
  repeated State states = 9;
}

message Population {
  int64 planet_id = 1;
  string allegiance = 2;
  string government = 3;
  string state = 4;
  string economy = 5;
  string security = 6;
  int64 population = 7;
  repeated Faction factions = 8;
  google.protobuf.Timestamp updated = 9;
}

message Powerplay {
  message Power {
    string name = 1;
    string state = 2;
  }

  int64 system_id = 1;
  Power power = 2;
  string allegiance = 3;
  string state = 4;
  string government = 5;
  google.protobuf.Timestamp updated = 9;
}
